src/huggingface_hub/hf_api.py

```diff
@@ -1761,6 +1897,67 @@ def upload_file(
                 " 0.11.0.",
                 FutureWarning,
             )
+        identical_ok = identical_ok if identical_ok is not None else True
+        if not identical_ok:
+            return self._legacy_upload_file(
+                path_or_fileobj=path_or_fileobj,
+                path_in_repo=path_in_repo,
+                repo_id=repo_id,
+                token=token,
+                repo_type=repo_type,
+                revision=revision,
+                identical_ok=identical_ok,
+                commit_message=commit_message,
+                commit_description=commit_description,
+                create_pr=create_pr,
+            )
+        commit_message = (
+            commit_message
+            if commit_message is not None
+            else f"Upload {path_in_repo} with huggingface_hub"
+        )
+        operation = CommitOperationAdd(
+            path_or_fileobj=path_or_fileobj,
+            path_in_repo=path_in_repo,
+        )
+
+        pr_url = self.create_commit(
+            repo_id=repo_id,
+            repo_type=repo_type,
+            operations=[operation],
+            commit_message=commit_message,
+            commit_description=commit_description,
+            token=token,
+            revision=revision,
+            create_pr=create_pr,
+        )
+        return (
+            pr_url
+            if pr_url is not None
+            else hf_hub_url(
+                repo_id, path_in_repo, revision=revision, repo_type=repo_type
+            )
+        )
+
+    def _legacy_upload_file(
+        self,
+        *,
+        path_or_fileobj: Union[str, bytes, BinaryIO],
+        path_in_repo: str,
+        repo_id: str,
+        token: Optional[str] = None,
+        repo_type: Optional[str] = None,
+        revision: Optional[str] = None,
+        identical_ok: bool = True,
+        commit_message: Optional[str] = None,
+        commit_description: Optional[str] = None,
+        create_pr: Optional[bool] = None,
+    ) -> str:
+        """
+        Deprecated - will be removed in 0.10.0
```

Comment:
This seems a bit inconsistent with `identical_ok` which is planned to be removed in 0.11.0, and whose value is used to determine whether or not to use this method in `upload_file`?